assets/js/components/PostSearcherAutoSuggest.js

```diff
@@ -67,6 +69,19 @@ export default function PostSearcherAutoSuggest( {
 		select( CORE_SITE ).getCurrentEntityTitle()
 	);
 
+	const postTitle = useRef( null );
+
+	const onFocus = useCallback( () => {
+		setIsActive( true );
+	}, [ setIsActive ] );
+
+	const onBlur = useCallback( () => {
+		setTimeout( () => {
+			setIsActive( false );
+			setSearchTerm( postTitle.current ?? currentEntityTitle ?? '' );
+		}, 100 );
```

Comment:
Why the delay here? 🤔

Reply:
When clicked on a result, the onBlur is fired before onSelect, setTimeout is there to make sure logic for onSelect runs first. That's the other reason why I had to use the useRef hook to get the updated post title from onSelectCallback.

Reply:
Could this maybe be done using an effect that runs when `isActive` changes rather than using a timeout?

Reply:
Hi, 

Tried to use `useEffect` and only set `isActive` on the `onBlur` handler. As the visibility of dropdown is dependent on isActive, the dropdown gets out of the way when a result is clicked, missing the onSelect handler completely. 

I have added a comment explaining the timeout. 

Cheers.